src/Datadog.Trace.ClrProfiler.Managed/Integrations/RabbitMQIntegration.cs

```diff
@@ -0,0 +1,900 @@
+        /// <param name="internal">The original internal setting</param>
+        /// <param name="nowait">The original nowait setting</param>
+        /// <param name="arguments">The original arguments setting</param>
+        /// <param name="opCode">The OpCode used in the original method call.</param>
+        /// <param name="mdToken">The mdToken of the original method call.</param>
+        /// <param name="moduleVersionPtr">A pointer to the module version GUID.</param>
+        [InterceptMethod(
+            TargetAssembly = RabbitMQAssembly,
+            TargetType = RabbitMQImplModelBase,
+            TargetMethod = "_Private_ExchangeDeclare",
+            TargetSignatureTypes = new[] { ClrNames.Void, ClrNames.String, ClrNames.String, ClrNames.Bool, ClrNames.Bool, ClrNames.Bool, ClrNames.Bool, ClrNames.Bool, ClrNames.Ignore },
+            TargetMinimumVersion = Major3Minor6Patch9,
+            TargetMaximumVersion = Major6)]
+        public static void ExchangeDeclare(
+            object model,
+            string exchange,
+            string type,
+            bool passive,
+            bool durable,
+            bool autoDelete,
+            bool @internal,
+            bool nowait,
+            object arguments,
+            int opCode,
+            int mdToken,
+            long moduleVersionPtr)
+        {
+            if (model == null) { throw new ArgumentNullException(nameof(model)); }
+
+            const string methodName = "_Private_ExchangeDeclare";
+            const string command = "exchange.declare";
+            Action<object, string, string, bool, bool, bool, bool, bool, object> instrumentedMethod;
+            var modelType = model.GetType();
+
+            try
+            {
+                instrumentedMethod =
+                    MethodBuilder<Action<object, string, string, bool, bool, bool, bool, bool, object>>
+                       .Start(moduleVersionPtr, mdToken, opCode, methodName)
+                       .WithConcreteType(modelType)
+                       .WithParameters(exchange, type, passive, durable, autoDelete, @internal, nowait, arguments)
+                       .WithNamespaceAndNameFilters(ClrNames.Void, ClrNames.String, ClrNames.String, ClrNames.Bool, ClrNames.Bool, ClrNames.Bool, ClrNames.Bool, ClrNames.Bool, ClrNames.Ignore)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: RabbitMQImplModelBase,
+                    methodName: methodName,
+                    instanceType: modelType.AssemblyQualifiedName);
+                throw;
+            }
+
+            RabbitMQTags tags = null;
+            using (var scope = CreateScope(Tracer.Instance, out tags, command, exchange: exchange))
```

Comment:
Style nit: Can we use a discard here?
```suggestion
            using (var scope = CreateScope(Tracer.Instance, out _, command, exchange: exchange))
```

Reply:
Discard added in 92957e00598a231e631db4348ee402f93ff6bdfc